capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -0,0 +1,129 @@
+    List<Club> clubs = new ArrayList<>();
+    for(Entity e : entities) {
+      String name = (String) e.getProperty("name");
+      String description = (String) e.getProperty("description");
+      String announcement = (String) e.getProperty("announcement");
+
+      @SuppressWarnings("unchecked") // Cast can't verify generic type.
+      Collection<String> posts = (ArrayList<String>) e.getProperty("posts");
+
+      @SuppressWarnings("unchecked") // Cast can't verify generic type.
+      Collection<Long> memberIDs = (HashSet<Long>) e.getProperty("memberIDs");
+      
+      @SuppressWarnings("unchecked") // Cast can't verify generic type.
+      Collection<Long> inviteIDs = (HashSet<Long>) e.getProperty("inviteIDs");
+
+      long ownerID = (long) e.getProperty("ownerID");
+      String gbookID = (String) e.getProperty("gbookID");
+      Club club = new Club(id, name, description, announcement, ownerID, gbookID);
+      clubs.add(club);
+    }
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(clubs));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setHeader("Access-Control-Allow-Methods", "GET");
+    response.setHeader("Access-Control-Allow-Credentials", "true");
+    response.setHeader("Access-Control-Allow-Origin", "https://3001-0b34ed39-12e2-4bb0-83f0-3edbd4365bbd.us-east1.cloudshell.dev");
+
+    String name = request.getParameter("name");
+    String description = request.getParameter("description");
+    long ownerID = Long.parseLong(request.getParameter("ownerID"));
+
+    String announcement = "";
+    if (request.getParameter("announcement") != null) {
+      announcement = request.getParameter("announcement");
+    }
+
+    Collection<Long> inviteIDs = new ArrayList<>();
+    if (request.getParameterValues("inviteIDs") != null) {
+      for(String s : request.getParameterValues("inviteIDs")) {
+        inviteIDs.add(Long.parseLong(s));
+      } 
+    }
+
+    String gbookID = "";
+    if (request.getParameter("gbookID") != null) {
+      gbookID = request.getParameter("gbookID");
+    }
+
+    Entity clubEntity = new Entity("Club");
+ 
+    clubEntity.setProperty("id", clubEntity.getKey().getId());
+    clubEntity.setProperty("name", name);
+    clubEntity.setProperty("description", description);
+    clubEntity.setProperty("announcement", announcement);
+    clubEntity.setProperty("posts" , new ArrayList<String>());
+    clubEntity.setProperty("ownerID", ownerID);
+    clubEntity.setProperty("inviteIDs", inviteIDs);
+    clubEntity.setProperty("memberIDs", new ArrayList<Long>());
+    clubEntity.setProperty("gbookID", gbookID);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(clubEntity);
+    
+  }
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Nice! I don't think we've used `doPut` before - was there a specific reason you wanted to have support for both `doPost` and `doPut` here?

Reply:
I wanted to allow the owner to change the name of the club, the book the club is reading, the club description etc without making a new club. I thought doPut made the most sense for this.